prototype/core/geo-json/src/main/java/org/eclipse/sensinact/gateway/geojson/GeoJsonObject.java

```diff
@@ -61,4 +63,44 @@ protected GeoJsonObject(GeoJsonType type) {
     public Map<String, Object> foreignMembers = new HashMap<>();
 
     public List<Double> bbox;
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(type, bbox);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof GeoJsonObject) {
+            final GeoJsonObject other = (GeoJsonObject) obj;
+            return type == other.type && Objects.equals(bbox, other.bbox)
+                    && Objects.equals(foreignMembers, other.foreignMembers);
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns the object description
+     */
+    protected abstract String getObjectDescription();
```

Comment:
```suggestion
    protected abstract String getObjectDescription(StringBuilder builder);
```

Passing a StringBuilder here would allow us to avoid creating lots of intermediate Strings.